packages/xstate-react/README.md

```diff
@@ -453,6 +453,74 @@ const Fetcher = ({ onResolve }) => {
 };
 ```
 
+### A note on external values
+
+When the machine options or context need to read from an external value, it's important that they can read the latest value in case it's changed! Therefore, using `withConfig` and `withContext` on the passed machine might end up being a recipe for bug. Instead, you can use one of the two approaches below:
+
+#### Pass a callback to `useInterpret` and `useMachine`
+
+```ts
+const machine = createMachine({
+  initial: 'foo',
+  context: {},
+  states: {
+    foo: {
+      on: {
+        CHECK: {
+          target: 'bar',
+          cond: 'hasOverflown'
+        }
+      }
+    },
+    bar: {}
+  }
+});
+
+const Component = () => {
+  const [id, setId] = useState(1);
+  // Passing a callback to `useMachine` makes sure the value of `id` is always up to date with the useState above
+  const [state, send] = useMachine(() =>
+    machine.withContext({ id: 1 }).withConfig({
+      guards: {
+        hasOverflown: () => id > 1
+      }
+    })
+  );
+};
```

Comment:
I don't think this is true - and I don't think we should recommend this either! The second option is both much cleaner, and soon will be perfectly typesafe.

Reply:
Yep - this isn't true, the `id` would still be closed over here. The lazy initialization in here only makes it slightly more apparent that this can be stale. If you want auto-updates you should use the second argument of the `useMachine`